forte/data/data_store.py

```diff
@@ -594,16 +666,90 @@ def _default_attributes_for_type(self, type_name: str) -> List:
         attr_dict: Dict = self._get_type_attribute_dict(type_name)
         attr_fields: Dict = self._get_entry_attributes_by_class(type_name)
         attr_list: List = [None] * len(attr_dict)
-        for attr_name, attr_id in attr_dict.items():
+        for attr_name, attr_info in attr_dict.items():
             # TODO: We should keep a record of the attribute class instead of
             # inspecting the class on the fly.
+            attr_id = attr_info[constants.ATTR_INDEX_KEY]
+
             attr_class = get_origin(attr_fields[attr_name].type)
             if attr_class in (FList, list, List):
                 attr_list[attr_id - constants.ATTR_BEGIN_INDEX] = []
             elif attr_class in (FDict, dict, Dict):
                 attr_list[attr_id - constants.ATTR_BEGIN_INDEX] = {}
         return attr_list
 
+    def _new_entry_types(
```

Comment:
what is `new` here in the function name?

Reply:
This is function is invoked when new entries are added to `_type_attributes` and thus their type information will also need to be added.

Reply:
1. Why is type_attribute information new when an entry is added? Some entries could have the same type
2. Still, I don't think anything is "new", the information is stored and get via `self._get_entry_attributes_by_class(type_name)`, so this function only `get` this information

Reply:
So this function is invoked in the case where it is the first time an entry of this type is being created (and thus it is not present in `_type_attributes`. But I do get your point, I think a better name for this function could be `fetch_entry_type_data`. Would this be okay?

Reply:
let's use the new name, since the "add" action seems to happen after this.

Reply:
Sounds good